### PR TITLE
#163426541 User Should be able to report an article

### PR DESCRIPTION
**Pull request by EfeAgare**
#### What does this PR do?
Have user should be able to report an article
Admin should be able to get one and all reports on an article

#### Description of Task to be completed?
Have the following endpoints working.
- User post a report on an Article    `POST /api/v1/article/:slug/reports`
- Admin gets all reports on an Article  `GET /api/v1/article/:slug/reports`
- Admin get all reports on all Articles `GET /api/v1/reports`

#### How should this be manually tested?
Clone this repo, cd into it and run npm install.
**With test suite:**
- Run `npm test`

**Using postman:**
- Run `npm run dev`,
- With postman, send requests to `localhost:3000/<api url>` with the relevant HTTP methods and request payloads.

#### What are the relevant pivotal tracker stories?
[#163426541](https://www.pivotaltracker.com/story/show/163426541)

### CODE REVIEW

**Review by biolabalo — DISMISSED** [review]

**Review by ozone4real — DISMISSED** [review]
LGTM

**Review by bobbylemm — DISMISSED** [review]

**Review by biolabalo — APPROVED** [review]

**Review by biolabalo — DISMISSED** [review]

File: server/controllers/reportController.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  /**
+   *@description controller method for getting all reports
+   * @param {object} req Request object
+   * @param {object} res Response object
+    * @param {Function} next passes control to the next middleware
+   * @returns {Object} a response object
+   */
+  static async getOneArticleReports(req, res, next) {
+    try {
+      const { slug } = req.params;
```
Comment (p-funky) [inline server/controllers/reportController.js:34]: I believe we have already concluded that we should be using `id` here and not slug?

Comment (EfeAgare) [reply server/controllers/reportController.js:34]: Yea, we didn't really conclude, as we said we will make more research on it. I then found from other resources and discuss with my teammates that changing the Article title implies changing the whole body of the article, meaning a new Article entirely. Also when titles of article are used as a slug, even if the title changes or is edited the slug remains the same. 
So we decided to still stick with the slug for now

**Review by bobbylemm — DISMISSED** [review]
LGTM

File: server/controllers/commentController.js [hunk 2]
```diff
@@ -27,14 +41,16 @@ class CommentController {
...
       }
-      req.body.articleId = article.dataValues.id;
-      const articleComments = await Comments.create(req.body);
+      const articleComments = await Comments.findAll({
+        where: { slug },
+        returning: true
+      });
       return res.status(201).json({
-        message: 'Comment has been posted successfully',
+        message: 'Comments retrieve successfully',
```
Comment (divinediscipline) [inline server/controllers/commentController.js]: This should be 'retrieved not retrieve'

**Review by omenkish — DISMISSED** [review]

File: server/tests/app/reportArticle.test.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    .get('/api/v1/reports')
+    .set('x-auth-token', adminVerifiedToken)
+    .then((res) => {
+      expect(res.status).to.equal(200);
+      expect(res.body.message).to.equal('Reports retrieve successful');
+    }));
+});
+
+describe('Welcome to Authors Haven', () => {
+  it('Should welcome users to Authors Haven', () => request(app)
```
Comment (divinediscipline) [inline server/tests/app/reportArticle.test.js]: Please check this test case. It appears to contradict itself.

File: server/tests/app/reportArticle.test.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    .set('x-auth-token', verifiedToken)
+    .then((res) => {
+      expect(res.status).to.equal(200);
+      expect(res.body.message).to.equal('welcome to authors haven platform');
+    }));
+});
+
+describe('Welcome to Authors Haven', () => {
+  it('Should welcome users to Authors Haven', () => request(app)
+    .get('/ai')
```
Comment (divinediscipline) [inline server/tests/app/reportArticle.test.js:109]: Please check this test case. It appears to contradict itself.

**Review by ozone4real — APPROVED** [review]
LGTM

**Review by biolabalo — APPROVED** [review]

**Pull request merged**